main.go

```diff
@@ -0,0 +1,86 @@
+package main
```

Comment:
I think the copyright transfer should be explicit with a header, in this and other files:

```
// Copyright 2019 Francisco Souza. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.
```